### ISSUE TITLE
Cannot download authorisation project code from docs example

### ISSUE THREAD

**Issue by deryckbeard** [dotnet/AspNetCore.Docs#19460]
All I wanted was to download the example code and work through the project, but cannot get the code.

The instructions to get the code, do not match the link provided

The link for the code took me into GIT hub with no zip download for the project.

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b66f8ffa-699c-8b11-0cab-805f5953805f
* Version Independent ID: 7c151ed5-fd78-84fd-a08a-88912990c4ca
* Content: [Create an ASP.NET Core app with user data protected by authorization](https://docs.microsoft.com/en-gb/aspnet/core/security/authorization/secure-data?view=aspnetcore-3.1)
* Content Source: [aspnetcore/security/authorization/secure-data.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/authorization/secure-data.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
@deryckbeard ... You have to back out to the top level ...

https://github.com/dotnet/AspNetCore.Docs

... and it's under the **Code** button there. That's the way GH works on all repos. The direct link in the ...

https://docs.microsoft.com/en-gb/aspnet/core/introduction-to-aspnet-core?view=aspnetcore-3.1#how-to-download-a-sample

... section will bring down the ZIP. That link is ...

https://codeload.github.com/dotnet/AspNetCore.Docs/zip/master

You can decompress and find the sample within the folder at the same path shown on the repo from there.

**Comment by deryckbeard** [comment]
Yes I worked that out Luke,

it took me some time and if the page and the links were constructed
correctly in the first instance I shouldn't have needed to.
 The download instruction link should have the detail of this email, and
the link for download should be the page where you download the file. As
you point out the Git hub page is pretty much useless for the article.

It is a difficult enough subject without having to do your own Sherlock
Holmes type investigation.

I take it you will reopen the case and close it once you have made the
changes to the document.

Best regards
Deryck

**Comment by guardrex** [comment]
> The instructions to get the code, do not match the link provided

They aren't supposed to match under the current approach. The instructions are to ...

1. Download the whole repo as a ZIP using that direct link in the *how to download* instructions (linked from the word "download").
2. Unzip and navigate to where the sample is located by looking at the destination of the sample link.

There has been an effort across all doc sets to have a individual sample links that should get a reader *just the sample* that they're looking for, but we never implemented it here ...

https://docs.microsoft.com/samples/browse/

@Rick-Anderson ... Are we still headed that way at some point ... to use that system?

**Comment by Rick-Anderson** [comment]
cc @scottaddie  @wadepickett

**Comment by deryckbeard** [comment]
Thanks for replying  Luke,

I don't care which approach you have taken as long as it is correctly
documented.
These are tutorials, they are supposed to help the user, not confuse and
make them angry.

I became extremely frustrated being led up the  garden path by the use of
incorrect links and then having to work out where and how I could access
the code sample.
Even more so when I googled it and found that the issue had been previously
reported and Rick patronisingly responded with in effect "tough, that's how
I've done it",
Github doesn't do that, and therefore  no help to the poor  person who
reported the issue. At least you explained how to get the code, for which I
am grateful.

Obviously it is better if it is consistent across all pages, but...
If you have used a non-standard approach, don't document it with the
standard approach; simply tell the reader what you have done and where to
find the code.
you have done it simply enough  here in the email, so  why not in the page?

BTW the tutorial is great, and very helpful (once I found the code :-)

Best regards
Deryck
